classes/condition.php

```diff
@@ -169,15 +169,14 @@ protected static function get_lang_string_keyword($completionstate) {
      *   this item
      */
     public function get_description($full, $not, \core_availability\info $info) {
-      global $DB;
-        // Get name for module.
-        $modc = get_courses();
+        global $DB;
 
-        $modname = get_string('missing', 'availability_othercompleted');
-        foreach ($modc as $modcs) {
-            if($modcs->id == $this->cmid){
-                $modname = $modcs->fullname;
-            }
```

Comment:
Ouch. Good improvement you've made to replace this.

Reply:
doing this in php seems harder than doing it in the db.... im a bit stumped :upside_down_face: